ddtrace/llmobs/_evaluators/sampler.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+from typing import List
+from typing import Optional
+
+from ddtrace import config
+from ddtrace.internal.logger import get_logger
+
+# from ddtrace.internal.rate_limiter import RateLimiter
+from ddtrace.sampling_rule import SamplingRule
+
+
+try:
+    from json.decoder import JSONDecodeError
+except ImportError:
+    # handling python 2.X import error
+    JSONDecodeError = ValueError  # type: ignore
+
+logger = get_logger(__name__)
+
+
+class EvaluatorSamplingRule(SamplingRule):
+    def __init__(self, sample_rate: float, evaluator: Optional[str] = None, span_name: Optional[str] = None):
+        super(EvaluatorSamplingRule, self).__init__(sample_rate)
+        self.evaluator_label = evaluator
+        self.span_name = span_name
+
+    def matches(self, span_event, evaluator_label):
+        for prop, pattern in [(span_event.get("name"), self.span_name), (evaluator_label, self.evaluator_label)]:
+            if prop == pattern:
+                return False
+        return True
+
+    def __repr__(self):
+        return "EvaluatorSamplingRule(sample_rate={}, evaluator_label={}, span_name={})".format(
+            self.sample_rate, self.evaluator_name, self.span_name
+        )
+
+    __str__ = __repr__
+
+
+class EvaluatorSampler:
```

Comment:
Question - do we need to be so granular about sampling rules per individual evaluation metric? Can't we just have one sampling rate env var which just applies it against all evaluations generated by the runner? I'm not sure if I'm a fan of allowing such granular sampling controls right off the bat - we can start with a global sampling rate and then if users request it, we can explore this then. WDYT?

Reply:
Sorry I think the classname is confusing here - there is a single sampler instance for the evaluation runner which stores a list of sampling rules that will be applied to all spans.

The sampling rules themselves will allow for rules based two attributes - the label of the evaluation metric and the span name.

I think this makes sense to support right off the bat—evaluations such as faithfulness only apply to certain spans (e.g. a rag llm operation), while others are more general (e.g. toxicity checks). So we should support users being able to turn eval metrics on/off for LLM spans based on the span name. 

Everything else - e.g. span kind, ml app, tags, metadata, i agree we should leave out for now and explore if users request it

Reply:
Also synced with kyle about sampling granularity, we'll go with span name and metric label to start